app/models/concerns/patient_reportable.rb

```diff
@@ -9,19 +9,21 @@ module PatientReportable
     scope :with_hypertension, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes).distinct }
     scope :excluding_dead, -> { where.not(status: :dead) }
 
-    scope :ltfu_as_of, ->(date) do
+    scope :ltfu_as_of, ->(time) do
+      ltfu_at = (time - LTFU_TIME).end_of_month
       joins("LEFT OUTER JOIN latest_blood_pressures_per_patient_per_months
              ON patients.id = latest_blood_pressures_per_patient_per_months.patient_id
-             AND #{sanitize_sql(["bp_recorded_at > ? AND bp_recorded_at < ?", (date.to_date - LTFU_TIME).end_of_month, date])}")
-        .where("bp_recorded_at IS NULL AND patients.recorded_at < ?", (date.to_date - LTFU_TIME).end_of_month)
+             AND #{sanitize_sql(["bp_recorded_at > ? AND bp_recorded_at < ?", ltfu_at, time])}")
+        .where("bp_recorded_at IS NULL AND patients.recorded_at <= ?", ltfu_at)
         .distinct(:patient_id)
     end
 
-    scope :not_ltfu_as_of, ->(date) do
+    scope :not_ltfu_as_of, ->(time) do
+      ltfu_at = (time - LTFU_TIME).end_of_month
       joins("LEFT OUTER JOIN latest_blood_pressures_per_patient_per_months
              ON patients.id = latest_blood_pressures_per_patient_per_months.patient_id")
         .where("bp_recorded_at > ? AND bp_recorded_at < ? OR patients.recorded_at >= ?",
-          ltfu_time_ago(date), date, ltfu_time_ago(date))
+          ltfu_at, time, ltfu_at)
```

Comment:
These are basically the same fixes as https://github.com/simpledotorg/simple-server/pull/2560, with the exception that we don't need to any UTC offset math.